src/betterproto/__init__.py

```diff
@@ -1051,6 +1289,46 @@ def parse(self: T, data: bytes) -> T:
             else:
                 setattr(self, field_name, value)
 
+            # If we have now loaded the expected length of the message, stop
+            if size is not None:
+                prev = read
+                read += len(parsed.raw)
+                if read == size:
+                    break
+                elif read > size:
+                    raise ValueError(
+                        f"Expected message of size {size}, can only read "
+                        f"either {prev} or {read} bytes - there is no "
+                        "message of the expected size in the stream."
+                    )
+
+        if size is not None and read < size:
+            raise ValueError(
+                f"Expected message of size {size}, but was only able to "
+                f"read {read} bytes - the stream may have ended too soon,"
+                " or the expected size may have been incorrect."
+            )
+
+        return self
+
+    def parse(self: T, data: bytes) -> T:
```

Comment:
```suggestion
    def parse(self: T, data: ReadableBuffer) -> T:
```
Would it be possible to change this now that it explicitly supports ReadableBuffer (from _typeshed)

Reply:
From what I understand, you wouldn't be able to simply change to using this, you would have to conditionally import from `_typeshed` and also conditionally overload/define the function based on `TYPE_CHECKING` - is this what you wanted me to do?

Reply:
I've made the change, though I'm not familiar with parts of this, so let me know if there's anything wrong with it